lib/RESTClient.js

```diff
@@ -351,9 +352,12 @@ class RESTClient {
                 res.on('data', data => {
                     ret.push(data);
                     retLen += data.length;
-                }).on('error', callback)
+                }).on('error', error => {
+                    log.error('error receiving response from metadata', { error });
+                    callbackOnce(errors.InternalError);
```

Comment:
You don't want to add an error description here to bring it up the callstack, The error should be printed by the called at one point ?

```suggestion
                    callbackOnce(errorInstances.InternalError.customizDescription(error.message));
```

Reply:
why not, although I'm not sure the error will actually be printed, but it shouldn't hurt to forward it.

Reply:
done (squashed in the main commit)